File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/module/project/view/parameters/ProjectMemberScreenData.java
package com.esofthead.mycollab.mobile.module.project.view.parameters;

import com.esofthead.mycollab.module.project.domain.ProjectMember;
import com.esofthead.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria;
import com.esofthead.mycollab.vaadin.mvp.ScreenData;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.5.2
 *
 */
public class ProjectMemberScreenData {
	public static class Search extends ScreenData<ProjectMemberSearchCriteria> {

		public Search(ProjectMemberSearchCriteria params) {
			super(params);
		}
	}

	public static class Add extends ScreenData<ProjectMember> {

		public Add(ProjectMember params) {
			super(params);
		}
	}

	public static class InviteProjectMembers extends ScreenData<Object> {

		public InviteProjectMembers() {
			super(null);
		}
	}

	public static class Read extends ScreenData<Object> {

		public Read(Object params) {
			super(params);
		}
	}
}

File: mycollab-mobile/src/main/scala/com/esofthead/mycollab/mobile/module/project/view/parameters/ProjectMemberScreenData.scala
package com.esofthead.mycollab.mobile.module.project.view.parameters

import com.esofthead.mycollab.module.project.domain.SimpleProjectMember
import com.esofthead.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria
import com.esofthead.mycollab.vaadin.mvp.ScreenData

/**
  * @author MyCollab Ltd
  * @since 5.2.5
  */
object ProjectMemberScreenData {

  class Search(param: ProjectMemberSearchCriteria) extends ScreenData[ProjectMemberSearchCriteria](param) {}

  class InviteProjectMembers() extends ScreenData[Any](null) {}

  class Read(param: Any) extends ScreenData[Any](param) {}

  class Edit(param: SimpleProjectMember) extends ScreenData[SimpleProjectMember](param) {}

}
